lib/grpc/status.ml

```diff
@@ -62,3 +62,25 @@ type t = { code : code; message : string option } [@@deriving show]
 let v ?message code = { code; message }
 let code t = t.code
 let message t = Option.map (fun message -> Uri.pct_encode message) t.message
+
+let extract_status headers =
+  let code, message =
+    match H2.Headers.get headers "grpc-status" with
+    | None -> (Unknown, Some "Expected gprc-status header, got nothing")
```

Comment:
In that case we could probe for the `:status` header and add it to the message if it's present, to make it more informative in the case of `Error`.

Reply:
You mean the HTTP2 `:status`? I think when it's not 200 we may not even want to try to interpret the response as gRPC as the spec requires a 200 status. See [my other PR about non-200 statuses](https://github.com/dialohq/ocaml-grpc/pull/22).